tests/foreman/api/test_repository.py

```diff
@@ -240,192 +294,244 @@ def test_positive_create_with_default_download_policy(self):
         default_dl_policy = entities.Setting().search(
             query={'search': 'name=default_download_policy'}
         )
-        self.assertTrue(default_dl_policy)
-        repo = entities.Repository(product=self.product, content_type='yum').create()
-        self.assertEqual(repo.download_policy, default_dl_policy[0].value)
+        assert default_dl_policy
+        assert repo.download_policy == default_dl_policy[0].value
 
     @pytest.mark.tier1
-    def test_positive_create_immediate_update_to_on_demand(self):
+    @pytest.mark.parametrize(
+        'repo_options', **parametrized([{'content_type': 'yum'}]), indirect=True
+    )
+    def test_positive_create_immediate_update_to_on_demand(self, repo):
         """Update `immediate` download policy to `on_demand`
         for a newly created YUM repository
 
         :id: 8a70de9b-4663-4251-b91e-d3618ee7ef84
 
+        :parametrized: yes
+
         :expectedresults: immediate download policy is updated to on_demand
 
         :CaseImportance: Critical
 
         :BZ: 1732056
         """
-        repo = entities.Repository(product=self.product, content_type='yum').create()
-        self.assertEqual(repo.download_policy, 'immediate')
+        assert repo.download_policy == 'immediate'
+
+        # Update repo 'download_policy' to 'on_demand'
         repo.download_policy = 'on_demand'
-        repo = repo.update(['download_policy'])
-        self.assertEqual(repo.download_policy, 'on_demand')
+        repo.update(['download_policy'])
```

Comment:
if you dont have `repo =`, won't you just be reading back from the object you just updated in line above?

Reply:
Fixed.